rpc/clients/typescript/test/ws_client_test.ts

```diff
@@ -1,507 +1,434 @@
-                const expectToBeCalledOnce = true;
-                const callback = callbackErrorReporter.reportNoErrorCallbackErrors(done, expectToBeCalledOnce)(
-                    async (orderEvents: OrderEvent[]) => {
-                        expect(orderEvents).to.have.length(1);
-                        expect(BigNumber.isBigNumber(orderEvents[0].signedOrder.makerAssetAmount)).to.equal(true);
-                        expect(BigNumber.isBigNumber(orderEvents[0].signedOrder.takerAssetAmount)).to.equal(true);
-                        expect(BigNumber.isBigNumber(orderEvents[0].signedOrder.makerFee)).to.equal(true);
-                        expect(BigNumber.isBigNumber(orderEvents[0].signedOrder.takerFee)).to.equal(true);
-                        expect(BigNumber.isBigNumber(orderEvents[0].signedOrder.salt)).to.equal(true);
-                        expect(BigNumber.isBigNumber(orderEvents[0].signedOrder.expirationTimeSeconds)).to.equal(true);
-                        expect(BigNumber.isBigNumber(orderEvents[0].fillableTakerAssetAmount)).to.equal(true);
-                        // tslint:disable-next-line:custom-no-magic-numbers
-                        expect(orderEvents[0].timestampMs).to.equal(new Date(timestamp).getTime());
 
-                        client.destroy();
-                    },
-                );
-                await client.subscribeToOrdersAsync(callback);
-            })().catch(done);
+                        done();
+                    });
+
+                    // Add orders to the mesh node.
+                    const ordersLength = 10;
+                    for (let i = 0; i < ordersLength; i++) {
+                        orders[i] = await orderFactory.newSignedOrderAsync({});
+                    }
+                    now = new Date(Date.now()).getTime();
+                    const validationResults = await deployment.client.addOrdersAsync(orders);
+                    expect(validationResults.accepted.length).to.be.eq(ordersLength);
+
+                    await subscription;
+                })().catch(done);
+            });
+
+            it('should receive subscription updates about cancelled orders', (done: DoneCallback) => {
+                (async () => {
+                    // Add an order and then cancel it.
+                    const order = await orderFactory.newSignedOrderAsync({});
+                    const validationResults = await deployment.client.addOrdersAsync([ order ]);
+                    expect(validationResults.accepted.length).to.be.eq(1);
+                    await exchange.cancelOrder(order).awaitTransactionSuccessAsync({ from: makerAddress });
+
+                    // Subscribe to order events and assert that only a single cancel event was received.
+                    const now = new Date(Date.now()).getTime();
+                    await deployment.client.subscribeToOrdersAsync((orderEvents: OrderEvent[]) => {
```

Comment:
It's not guaranteed that you'll receive events if thdey occurred before you subscribe. Let's therefore subscribe before we cancel the order.